frontend/src/services/HttpService.ts

```diff
@@ -0,0 +1,352 @@
+/* eslint-disable max-classes-per-file */
```

Comment:
This rule is there for a reason. At least move the exported classes to separate files.

Put everything in a single file under a folder `HttpService`.
Create `index.ts` with the content like
```ts
export { HttpService } from './HttpService';
export { HttpRequest } from './HttpRequest';
...
```

and use the imports like that:
```ts
import { HttpService, HttpRequest } from './services/HttpService'
```

Looks the same from the outside, but much cleaner if you look into the files.